torch/csrc/distributed/rpc/python_rpc_handler.h

```diff
@@ -61,6 +61,24 @@ class PYBIND11_EXPORT PythonRpcHandler {
   PythonRpcHandler();
   ~PythonRpcHandler() = default;
 
+  class GilWaitTimeGuard {
+   public:
+    explicit GilWaitTimeGuard();
+    void markAcquired();
+
+   private:
+    std::chrono::time_point<std::chrono::high_resolution_clock> start_;
+  };
+
+// A macro that grabs the GIL, profiling the acquisition time. The average GIL
+// acquisition time will be recorded in RpcAgent's getMetrics().
+#define PROFILE_GIL_SCOPED_ACQUIRE                             \
+  GilWaitTimeGuard g;                                          \
+  pybind11::gil_scoped_acquire ag;                             \
+  if (RpcAgent::getDefaultRpcAgent()->getMetricsProfiling()) { \
+    g.markAcquired();                                          \
+  }
```

Comment:
Just realized this is not really a guard since we don't actually do anything when the object is destroyed. Can we change the macro to do this instead:

```
std::chrono::time_point<std::chrono::high_resolution_clock> startTime;
bool should_profile_gil = RpcAgent::getDefaultRpcAgent()->getMetricsProfiling();
if (should_profile_gil) {
  startTime = now();
}
pybind11::gil_scoped_acquire ag;
if (should_profile_gil) {
  auto gil_time = std::chrono::duration_cast<std::chrono::microseconds>(
      std::chrono::high_resolution_clock::now() - start);
  RpcAgent::getDefaultRpcAgent()->addGilWaitTime(gil_time);
}
```

This would minimize the overhead when GIL profiling is disabled and we probably don't need a separate GilWaitTimeGuard class since the macro handles this.